src/VisualStudio/Core/Next/Remote/JsonRpcSession.cs

```diff
@@ -65,9 +65,23 @@ internal class JsonRpcSession : RemoteHostClient.Session
 
         private async Task InitializeAsync()
         {
-            // all roslyn remote service must based on ServiceHubServiceBase which implements Initialize method
-            await _snapshotClient.InvokeAsync(WellKnownServiceHubServices.ServiceHubServiceBase_Initialize, _currentSessionId, PinnedScope.SolutionChecksum.ToArray()).ConfigureAwait(false);
-            await _serviceClient.InvokeAsync(WellKnownServiceHubServices.ServiceHubServiceBase_Initialize, _currentSessionId, PinnedScope.SolutionChecksum.ToArray()).ConfigureAwait(false);
+            CancellationToken.ThrowIfCancellationRequested();
+
+            try
+            {
+                // all roslyn remote service must based on ServiceHubServiceBase which implements Initialize method
+                await _snapshotClient.InvokeAsync(WellKnownServiceHubServices.ServiceHubServiceBase_Initialize, _currentSessionId, PinnedScope.SolutionChecksum.ToArray()).ConfigureAwait(false);
```

Comment:
underlying issue is we are using jsonRpc.InvokeAsync which doesnt accept cancellation and we created our own cancellation mechanism around it. so regardless where we put it, whoever deal with JsonRpcStream directly need to know about this issue.

fixing root cause is using buildin cancellation mechanism newer version of (preview 5) JsonRpcStream supports. but that means each JsonRpc.InvokeAsync accepting cancellation token rather than session owning it like we do now.